src/main/scala/pricemigrationengine/services/CohortTable.scala

```diff
@@ -1,19 +1,69 @@
 package pricemigrationengine.services
 
+import java.util
+
+import com.amazonaws.services.dynamodbv2.model.{AttributeValue, QueryRequest}
+import pricemigrationengine.dynamodb.{DynamoDBZIO, DynamoDBZIOError}
 import pricemigrationengine.model._
-import zio.ZIO
+import zio.stream.ZStream
+import zio.{IO, UIO, ZIO, ZLayer}
+
+import scala.jdk.CollectionConverters._
+
 
 object CohortTable {
 
   trait Service {
-    def fetch(filter: CohortTableFilter, batchSize: Int): ZIO[Any, CohortFetchFailure, Set[CohortItem]]
+    def fetch(
+      filter: CohortTableFilter,
+      batchSize: Int
+    ): IO[CohortFetchFailure, ZStream[Any, CohortFetchFailure, CohortItem]]
 
     def update(result: EstimationResult): ZIO[Any, CohortUpdateFailure, Unit]
   }
 
-  def fetch(filter: CohortTableFilter, batchSize: Int): ZIO[CohortTable, CohortFetchFailure, Set[CohortItem]] =
+  def fetch(filter: CohortTableFilter, batchSize: Int): ZIO[CohortTable, CohortFetchFailure, ZStream[Any, CohortFetchFailure, CohortItem]] =
     ZIO.accessM(_.get.fetch(filter, batchSize))
 
   def update(result: EstimationResult): ZIO[CohortTable, CohortUpdateFailure, Unit] =
     ZIO.accessM(_.get.update(result))
+
+  val impl: ZLayer[DynamoDBZIO, Nothing, UserRepo] =
```

Comment:
Perhaps keep the implementation separate from the interface so that it's clearer to distinguish?